airbyte-commons/src/main/java/io/airbyte/commons/util/AutoCloseableIterators.java

```diff
@@ -72,6 +72,12 @@
     return new DefaultAutoCloseableIterator<>(stream.iterator(), stream::close);
   }
 
+  public static <T> List<T> toList(AutoCloseableIterator<T> iterator) throws Exception {
```

Comment:
Can you add a comment to expose that it closes the iterator?

Reply:
might be clearer to name this function something like `toListAndClose`.

Reply:
yup